ui/js/directives/treeherder/clonejobs.js

```diff
@@ -166,87 +177,173 @@ treeherder.directive('thCloneJobs', [
-                }
+                //Keep track of visibility with this property. This
+                //way down stream job consumers don't need to repeatedly
+                //call showJob
+                job.visible = thJobFilters.showJob(job);
 
-                hText = getHoverText(job);
-                key = getJobMapKey(job);
+                addJobBtnToArray(job, lastJobSelected, jobBtnArray);
+            }
+            jobList.append(jobBtnArray);
+        };
 
-                jobStatus = thResultStatusInfo(resultState);
+        var addJobBtnToArray = function(job, lastJobSelected, jobBtnArray) {
+            var jobStatus, jobBtn;
 
-                //Add a visual indicator for a failure classification
-                jobStatus.key = key;
-                if(parseInt(job.failure_classification_id, 10) > 1){
-                    jobStatus.value = job.job_type_symbol + '*';
-                    if (jobStatus.btnClassClassified) {
-                        // For result types that are displayed more prominently
-                        // when unclassified, switch to the more subtle classified
-                        // style.
-                        jobStatus.btnClass = jobStatus.btnClassClassified;
-                    }
-                } else {
-                    jobStatus.value = job.job_type_symbol;
-                }
+            jobStatus = thResultStatusInfo(thResultStatus(job), job.failure_classification_id);
+            jobStatus.key = getJobMapKey(job);
+            jobStatus.value = job.job_type_symbol;
+            jobStatus.title = getHoverText(job);
+            jobBtn = $(jobBtnInterpolator(jobStatus));
+
+            //If the job is currently selected make sure to re-apply
+            //the job selection styles
+            if( !_.isEmpty(lastJobSelected.job) &&
+                (lastJobSelected.job.id === job.id)){
+
+                setSelectJobStyles(jobBtn);
+
+                //Update the selected job element to the current one
+                ThResultSetStore.setSelectedJob($rootScope.repoName, jobBtn, job);
+            }
+            showHideElement(jobBtn, job.visible);
+
+            jobBtnArray.push(jobBtn);
+            // add a zero-width space between spans so they can wrap
+            jobBtnArray.push(' ');
+        };
+
+        var getGroupInfo = function(el, groupMap) {
+            var gi = {};
+            try {
+                gi.platformGroupEl = $(el).closest(".platform-group");
+                gi.grpJobList = gi.platformGroupEl.find(".group-job-list");
+                gi.grpCountList = gi.platformGroupEl.find(".group-count-list");
+                gi.key = gi.platformGroupEl.find(".job-group").attr(groupKeyAttr);
+                gi.jgObj = groupMap[gi.key].grp_obj;
+                return gi;
+            } catch(TypeError) {
+                return null;
+            }
+        };
+
+        /**
+         * Group most resultStates as just counts.  Keep "failed" as job-btns
```

Comment:
Maybe we can be a bit more descriptive here for folks who come across this part of the code, but not sure what the optimal wording might be? Perhaps something like

```
Group non-failed jobs as '+n' counts in the UI by default, and failed jobs as individual buttons.
Each job receives a corresponding resultState which determines its display.
```

Or whatever makes sense to you guys :)